### ISSUE TITLE
Using enum for index in an array of methodmaps causes tag mismatch

### ISSUE THREAD

**Issue by florinilie139** [alliedmodders/sourcepawn#745]
I don't know if this was posted before or not but this is my issue:

I used enum structs to make generic C++ like structure called GameStruct, and I have another enum called SpecialDays with names, like ZombieDay, Gunday etc that have int values.

I store all the GameStruct different configurations into an array, like this GameStruct g_GameList[_:MaxSpecialDays] and g_GameList[ZombieDay].Initialize("InitializationOfZombieDay") which is a function with a lot of parameters
No warning whatsoever.

Recently, for helping with my productivity and making things easier to read (and for my mental sake) I changed the enum struct to a methodmap that inherits Dynamic, something like a class 

Now, the same line that had g_GameList[ZombieDay].Initialize is now g_GameList[ZombieDay] = GSZombieDay() where all the initialization is done in the "constructor". Now this line gives me a warning that I have a tag mismatch. If I change the ZombieDay enum to 1, it doesn't give me a warning anymore. I tried adding view as int or _: before the enum, still gives me this warning. The code itself works no problem, but I don't want to get spammed with a lot of warnings, every time I compile the plugin.

I am using SourcePawn Compiler 1.10.0.6515 and I tried with 6528 as well

Thanks, here is my warning list
![image](https://user-images.githubusercontent.com/58662347/145803581-ae0cd26b-69e2-45d2-a7b1-2cf5bcadd72c.png)

**Comment by Headline** [comment]
Hi thanks for reporting this!

I'm finding it a little hard to follow exactly what's happening here, is there any chance you can post a small proof-of-concept code sample that demonstrates this unexpected output?

**Comment by florinilie139** [comment]
Sure, I will try to post the before and after the change to the method map.

Before
`
enum SpecialDays{
    NormalDay,        //0
    ZombieDay,        
    MaxSpecialDays
}

enum struct GameStruct{
    SpecialDays DayNr; 
    //Somebunch of variables
    
    void Initialize(SpecialDays _nr, char[] _name, ...)
    {
          //initializing code that initializes the "Somebunch of variables"
    }
}

GameStruct g_GameList[MaxSpecialDays]; 

public void OnAllPluginsLoaded()
{
    //void Initialize (SpecialDays _nr, char[] _name)
    g_GameList[ZombieDay].Initialize(ZombieDay, "Zombie Day") //no warning for [ZombieDay]
`

After, 
// test.sp(55) : warning 213: tag mismatch
// test.sp(60) : warning 213: tag mismatch
[test.txt](https://github.com/alliedmodders/sourcepawn/files/7703713/test.txt)

**Comment by Fyren** [comment]
If you use an enumerator to index an array, that whole expression's tag is the enum.  Whether you do something like `array[ZombieDay]` or `array[_:ZombieDay]` or even `array[view_as<int>(ZombieDay)]`, then it's like doing `SpecialDays:array[ZombieDay]`.  It's how the old-syntax enum structs work.

You can either use 1.11, with 1.10 define your enum as `enum SpecialDays:{ ... }` (note the colon), or with 1.10 I guess do `sometag:array[ZombieDay]`.
